lighthouse-cli/test/fixtures/byte-efficiency/tester.html

```diff
@@ -90,41 +91,63 @@ <h2>Byte efficiency tester page</h2>
   <!-- FAIL(responsive): image is 50% used at DPR 3 -->
   <!-- PASS(responsive-inverse): image is big enough -->
   <!-- PASS(offscreen): image is onscreen -->
+  <!-- PASS(unsized-images): has CSS sizing -->
   <img style="width: 256px; height: 170px;" src="lighthouse-1024x680.jpg">
 
   <!-- PASS(optimized): image is JPEG optimized -->
   <!-- PASS(webp): image is WebP optimized -->
   <!-- FAIL(responsive): image is 25% used at DPR 2.625 -->
   <!-- PASS(responsive-inverse): image is big enough -->
   <!-- PASS(offscreen): image is offscreen but lazily loaded -->
+  <!-- PASS(unsized-images): CSS sizing is valid with current checks -->
   <img style="width: 0px; height: 0px;" src="lighthouse-2048x1356.webp?size0" loading="lazy">
 
   <!-- PASS(optimized): image is JPEG optimized -->
   <!-- FAIL(webp): image is not WebP optimized -->
   <!-- FAIL(responsive): image is not fully used at DPR 2.625 -->
   <!-- PASS(responsive-inverse): image is big enough -->
   <!-- PASS(offscreen): image is onscreen -->
+  <!-- PASS(unsized-images): has CSS sizing -->
   <img style="width: 160px; height: 110px;" src="lighthouse-480x320.jpg">
 
+  <!-- PASS(optimized): image is JPEG optimized -->
+  <!-- FAIL(webp): image is not WebP optimized -->
+  <!-- FAIL(responsive): image is not fully used at DPR 2.625 -->
+  <!-- PASS(responsive-inverse): image is big enough -->
+  <!-- PASS(offscreen): image is onscreen -->
+  <!-- PASS(unsized-images): has attribute sizing -->
+  <img width="160" height="110" src="lighthouse-480x320.jpg">
+
   <!-- PASS(optimized): image is JPEG optimized -->
   <!-- PASS(webp): image has insigificant WebP savings -->
   <!-- PASS(responsive): image is used at full size -->
   <!-- FAIL(responsive-inverse): image does not account for DPR 2.625 -->
   <!-- PASS(offscreen): image is onscreen -->
+  <!-- FAIL(unsized-images): no sizing information -->
   <img src="lighthouse-320x212-poor.jpg">
 
+  <!-- PASS(optimized): image is JPEG optimized -->
+  <!-- PASS(webp): image has insigificant WebP savings -->
+  <!-- PASS(responsive): image is used at full size -->
+  <!-- FAIL(responsive-inverse): image does not account for DPR 2.625 -->
+  <!-- PASS(offscreen): image is onscreen -->
+  <!-- FAIL(unsized-images): invalid CSS sizing -->
+  <img style="width: auto; height: auto;" src="lighthouse-320x212-poor.jpg">
+
   <!-- PASS(optimized): image is JPEG optimized -->
   <!-- PASS(webp): image is WebP savings -->
   <!-- PASS(responsive): image is used at full size with srcset -->
   <!-- PASS(responsive-inverse): image is big enough -->
   <!-- PASS(offscreen): image is onscreen -->
+  <!-- FAIL(unsized-images): sizing available describes viewport, not the image -->
```

Comment:
hm this one is absolutely positioned though, I would expect us to pass it for the purposes of CLS

Reply:
I'm a bit confused because so far unsized-images' scope is only focused on explicit sizing & so I feel it would make more sense to check for absolute positioning in a separate audit? Can we justify that there are merits to have explicitly sized images regardless of absolute positioning?

& this [article](https://web.dev/optimize-cls/#modern-best-practice) made me think that if img elements use srcset, they should still have explicit width and height sizing:

> srcset defines the images you allow the browser to select between and what size each image is. To ensure <img> width and height attributes can be set, each image should use the same aspect ratio

Additionally for the picture element example with absolute positioning

> (re: picture elements) It's very possible these images could have different aspect ratios and browsers are still evaluating what the most efficient solution here should be, including if dimensions should be specified on all sources. Until a solution is decided on, relayout is still possible here.

let me know if I am misunderstanding something!

Reply:
> so I feel it would make more sense to check for absolute positioning in a separate audit

I'm not sure what such an audit would be checking for or what concern it's representing. Could you explain a bit what splitting it up might look like?

>  Can we justify that there are merits to have explicitly sized images regardless of absolute positioning?

I'm not aware of what they are at the moment, so maybe this just needs commenting about why we still care :) 

My current understanding of the reason behind this audit is that we care about CLS and when an image lacks explicit sizing, it is at risk for being resized and thus reflowing content *around* it such that it might shift. 

If an image is absolutely positioned (or `position: fixed`) then there is no concern about layout shifts because no other element's position can shift as a result of this element being resized. i.e. the developer has manually done the space reservation that its UI requires.

(Also note that if we end up considering `position: absolute` to not be sufficient than we will need to consider `top`/`left`/`bottom`/`right` as explicit size, I'm hoping we can skip this entirely 😃 )

> this article made me think that if img elements use srcset, they should still have explicit width and height sizing:

Totally right there, srcset just controls what asset is being used, not how large the display size might be 👍 

> Additionally for the picture element example with absolute positioning

Right again being `<picture>` doesn't escape it from needing explicit size, just the fact that `class="onscreen"` gives it absolute positioning.

Reply:
> I'm not sure what such an audit would be checking for or what concern it's representing. Could you explain a bit what splitting it up might look like?

I was thinking about how when future CLS audits are created i.e. (videos, iframes, etc.) that we would be redundantly checking for `position: absolute` but now I realize that isn't really an issue lol 

However, in relation to my earlier comment about the 'scope' of the unsized-images audit, does it matter that it currently isn't clear that we'll checking for `position: absolute`? i.e. do we need to add to the description of this audit? 

I'm imagining there is a user that passes the unsized-images test without using width/height anywhere because they use absolute positioning for all their images & so they're confused about why we're saying that all their images have explicit height & width

Reply:
> does it matter that it currently isn't clear that we'll checking for position: absolute

In general, no. There are many situations in Lighthouse where the audit's advice just doesn't apply if XYZ situation happens. We don't have to include *all* of those situations in the audit title or name or even the description if it's not the common or expected mediation. In these cases, we tend to just identify them in a list in the documentation article if we mention them at all. 

There's a strong argument for *not* mentioning the `absolute` position in the report because it's an action we really don't want developers to take to fix this issue. It's just that if they already have an image that's `position: absolute`, we wouldn't want to falsely flag it as needing explicit size to reduce CLS.

To the heart of your point, the true purpose of this audit might be better served with a title like "Image elements can shift content on load", but because the reason this happens is overwhelmingly that width/height are missing, that's the way we've framed it for the Lighthouse audience to make the action item clearer without clicking through to docs.

> do we need to add to the description of this audit?

We already have "to reduce layout shifts and improve CLS" so those who know about absolute positioning effects won't be surprised there. The primary tweak we might want to make is remove "Always" at the beginning of the situation since that's not totally true :)

> that so far only talks to the user about "explicit width and height"

Not entirely true ;) See above, but we describe the audit's purpose "to reduce layout shifts and improve CLS". If I assume the tool telling me this is reasonable, that implies unsized images that don't impact layout shifts or CLS won't be flagged.

Reply:
thanks for the explanation! I'm going to double check which other `position` attributes might be relevant & then I'll collect/process them in `image-elements` & `unsized-images` in a way similar to height/width

Reply:
Great idea! luckily for us getting position is more easily done with getComputedStyle. We don't really care who or what sets it since the browser won't "compute" it to be something we would ignore :)

Reply:
ah right! thanks for the heads up!